src/_pytest/config/__init__.py

```diff
@@ -1305,6 +1316,69 @@ def _getconftest_pathlist(
             values.append(relroot)
         return values
 
+    def _get_append_ini_values(self, name: str, ini_type) -> List[str]:
+        value = []
+        # append_ini is a list of "ini=value" options
+        # append all values if multiple values are set for same ini-name,
+        # e.g. -a foo=bar1 -a foo=bar2 will append 'bar1 bar2' to foo
+        for ini_config in self._append_ini:
+            try:
+                key, append_ini_value = ini_config.split("=", 1)
+            except ValueError:
+                raise UsageError(
+                    "-a/--append-ini expects option=value style (got: {!r}).".format(
+                        ini_config
+                    )
+                )
+            else:
+                if key == name:
+                    if ini_type in ["pathlist", "args"]:
+                        value.append(append_ini_value)
+                    elif ini_type == "linelist":
+                        value.append("\n{}".format(append_ini_value))
+                    else:
+                        self._warn_or_fail_if_strict(
+                            "append_ini option invalid for argument '{}' with type '{}'".format(
+                                name, ini_type
+                            )
+                        )
+        return value
+
+    def _resolve_ini_option(
+        self,
+        name: str,
+        default_value: str,
```

Comment:
This is not necessarily a string; it has the type corresponding to the arg type (`"bool"` -> `bool` etc.), so probably just leave it untyped for now until we get around to clarifying things.